internal/credentials/credentials.go

```diff
@@ -86,21 +87,58 @@ type CredentialsService interface {
 	Get(guid string) (*Credential, error)
 	List() ([]Credential, error)
 	Set(name string, url string, ak string) (*Credential, error)
+	Reset() error
 }
 
 // The main credentials service constructor that determines if the system's keyring is available to be used,
 // if not, returns a file based credentials service.
+// Additionally, tries to load the current credentials to handle resetting in case of data corruption.
 func NewCredentialsService(log logging.Logger) (CredentialsService, error) {
 	krService := NewKeyringCredentialsService(log)
 	if krService.IsSupported() {
-		return krService, nil
+		err := enforceKeyringIntegrity(krService, log)
+		return krService, err
 	}
 
 	log.Debug("Fallback to file managed credentials service due to unavailable system keyring")
 	fcService, err := NewFileCredentialsService(log)
 	if err != nil {
-		return nil, types.NewAgentError(types.ErrorCredentialServiceUnavailable, err, nil)
+		return fcService, enforceFilecredsIntegrity(fcService, err, log)
```

Comment:
I was a bit confused how the credentials were being reset right when the extension activated, but if I'm understanding correctly the enforcement happens before the APIs are even called since it occurs during the setup of the Credentials Service.

Perhaps we can do something with a different flow to ensure that notifications are always present?

Some potential ideas (these are really quick so take with a large grain of salt):
- have a `POST /api/credentials/reset` endpoint that we can call when we get certain errors
- only call the enforcement when the APIs are actually called